sgx-hasher/src/lib.rs

```diff
@@ -79,6 +79,55 @@ impl Hasher {
     }
 }
 
+fn bn_to_rsanum_inner(bn: bn::BigNum) -> Result<[u8; 384], openssl::error::ErrorStack> {
+    let mut array = [0u8; 384];
+    array.copy_from_slice(&bn.to_vec());
+    array.reverse();
+    Ok(array)
+}
```

Comment:
This function never appears to return an error. Also, `copy_from_slice()` panics if the array size isn't correct. We definitely don't want to panic on user input. So you need a manual test to see if the `vec` is the right length and return an error if it isn't.

Reply:
I put in a manual check here.